tests/unit/actor/test_actor_create_request_list.py

```diff
@@ -0,0 +1,204 @@
+        )
+    )
+
+    example_start_urls_input = [
+        {'requests_from_url': 'https://crawlee.dev/file.txt', 'method': 'GET'},
+        {'url': expected_simple_url, 'method': 'GET'},
+        {'requests_from_url': 'https://www.crawlee.dev/file2', 'method': 'GET'},
+    ]
+
+    http_client = HttpxHttpClient()
+    with mock.patch.object(http_client, 'send_request', return_value=_create_dummy_response(response_bodies)):
+        generated_request_list = await Actor.create_request_list(
+            actor_start_urls_input=example_start_urls_input, http_client=http_client
+        )
+        generated_requests = []
+        while request := await generated_request_list.fetch_next_request():
+            generated_requests.append(request)
+
+    # Check correctly created requests' urls in request list
+    assert {generated_request.url for generated_request in generated_requests} == expected_urls
+
+async def test_actor_create_request_list_from_url_additional_inputs()  -> None:
+    """Test that all generated request properties are correctly populated from input values."""
+    expected_simple_url = 'https://www.someurl.com'
+    example_start_urls_input = [
+        {'requests_from_url': 'https://crawlee.dev/file.txt',             'method': 'POST',
+            'headers': {'key': 'value'},
+            'payload': 'some_payload',
+            'user_data': {'another_key': 'another_value'}},
+    ]
+    response_bodies = iter((expected_simple_url,))
+    http_client = HttpxHttpClient()
+    with mock.patch.object(http_client, 'send_request', return_value=_create_dummy_response(response_bodies)):
+        generated_request_list = await Actor.create_request_list(
+            actor_start_urls_input=example_start_urls_input, http_client=http_client
+        )
+        request = await generated_request_list.fetch_next_request()
+
+    # Check all properties correctly created for request
+    assert request.url == expected_simple_url
+    assert request.method == example_start_urls_input[0]['method']
+    assert request.headers.root == example_start_urls_input[0]['headers']
+    assert request.payload == example_start_urls_input[0]['payload'].encode('utf-8')
+    expected_user_data = UserData()
+    for key, value in example_start_urls_input[0]['user_data'].items():
+        expected_user_data[key] = value
+    assert request.user_data == expected_user_data
+
+
+@pytest.mark.parametrize('true_positive', [
+    'http://www.something.com',
+    'https://www.something.net',
+    'http://nowww.cz',
+    'https://with-hypen.com',
+    'http://number1.com',
+    'http://www.number.123',
+    'http://many.dots.com',
+    'http://a.com',
+    'http://www.something.com/somethignelse'
+    'http://www.something.com/somethignelse.txt',
+    #  "http://non-english-chars-á.com"  # re module not suitable, regex can do this with \p{L}. Do we want this?
```

Comment:
JS version uses \p{L} and thus it supports this use case. Should we port it to Python as well? It would require using custom **regex** package instead of Python standard **re** package.

JS implementation:
https://github.com/apify/crawlee/blob/master/packages/utils/src/internals/general.ts#L9

Python regex package that supports \p{L} syntax:
https://github.com/mrabarnett/mrab-regex

Reply:
1. if the package is not huge in terms of disk space, there's no problem in adding it :shrug: 
2. **however**, if `\p{L}` is the only thing that's missing, I believe that `\w` does the same in Python, doesn't it?

Reply:
**\p{L}** should match just letters, not numbers or underscore and it should match non-English letters like á, í, é, ...as well
I think that **\w** is basically matching **[a-zA-Z0-9_]** , so  does not include letters like á, í, é, ...

Reply:
> however, if \p{L} is the only thing that's missing, I believe that \w does the same in Python, doesn't it?

`\w` is not the same as `\p{L}`, but I wouldn't care about this if its not natively supported by python.

Reply:
Quoting https://docs.python.org/3/howto/regex.html#matching-characters,

> Let’s take an example: \w matches any alphanumeric character. If the regex pattern is expressed in bytes, this is equivalent to the class [a-zA-Z0-9_]. If the regex pattern is a string, \w will match all the characters marked as letters in the Unicode database provided by the [unicodedata](https://docs.python.org/3/library/unicodedata.html#module-unicodedata) module. You can use the more restricted definition of \w in a string pattern by supplying the [re.ASCII](https://docs.python.org/3/library/re.html#re.ASCII) flag when compiling the regular expression.

So... good enough?

Reply:
So is that particular test with `http://non-english-chars-á.com` commented out for no reason and it will work fine? I guess not.

Either way, even if it's not working, I would say it is good enough to ignore this difference for now and only deal with that once someone actually tries to work with such a domain and asks us to fix it.